packages/host/app/services/matrix-service.ts

```diff
@@ -324,6 +327,10 @@ export default class MatrixService extends Service {
         console.log('Error starting Matrix client', e);
         await this.logout();
       }
+
+      if (onAfterStart) {
```

Comment:
How is adding this callback better than awaiting the call to `start` and running something after it?

Reply:
the call so start lives in task within a component.  the act of logging in sets the tracked property `MatrixService.isLoggedIn` which is consumed in the container.gts  that you commented about earlier. when that is set to `true` the login component that called the start is actually torn down. within the EC task, the `awaits` are actually `yields`, nothing is evaluated after we yield to this.matrixService.start() in the login.gts component since the matrix services sets a tracked property that destroys the login component that yielded to  this.matrixService.start()